client/web/src/repo/components/RepoHeaderActions/RepoHeaderActions.tsx

```diff
@@ -57,3 +67,59 @@ export const RepoHeaderActionAnchor = React.forwardRef((props: RepoHeaderActionA
     )
 }) as ForwardReferenceComponent<typeof ButtonLink, RepoHeaderActionAnchorProps>
 RepoHeaderActionAnchor.displayName = 'RepoHeaderActionAnchor'
+
+type RepoHeaderActionMenuLinkProps = MenuLinkProps & {
+    /**
+     * to determine if this button is for file or not
+     */
+    file?: boolean
+    className?: string
+}
+
+export const RepoHeaderActionMenuLink = React.forwardRef((props: RepoHeaderActionMenuLinkProps, reference) => {
+    const { children, className, file, ...rest } = props
+
+    return (
+        <MenuLink
+            className={classNames(file ? styles.fileAction : styles.action, className)}
+            ref={reference}
+            // keep Menu open on select (mouse only)
+            onMouseUp={event => event.preventDefault()}
+            {...rest}
+        >
+            {children}
+        </MenuLink>
+    )
+}) as ForwardReferenceComponent<'a', RepoHeaderActionMenuLinkProps>
+RepoHeaderActionMenuLink.displayName = 'RepoHeaderActionMenuLink'
+
+export type RepoHeaderActionMenuItemProps = MenuItemProps & {
+    /**
+     * to determine if this anchor is for file or not
+     */
+    file?: boolean
+    className?: string
+}
+
+export const RepoHeaderActionMenuItem = React.forwardRef((props: RepoHeaderActionMenuItemProps, reference) => {
+    const { children, className, file, onSelect, ...rest } = props
+
+    return (
+        <MenuItem
+            className={classNames(file ? styles.fileAction : styles.action, className)}
+            ref={reference}
+            // keep Menu open on select (mouse only)
+            onMouseUp={event => event.preventDefault()}
+            // we still need `onSelect` binding for keyboard users
+            // but we don't know how to keep Menu open in this case
+            onSelect={onSelect}
+            // `onClick` will be triggered when using mouse
+            // instead of `onSelect` because we already catch `onMouseUp`
+            onClick={onSelect}
```

Comment:
```suggestion
            onSelect={onSelect}
```

I think we can do this and happily close the menu on click AND on keyboard select. WDYT?